backend/src/impl/db_utils/system_db_utils.py

```diff
@@ -146,11 +147,24 @@ def find_systems(
             filt["creator"] = creator
         if shared_users:
             filt["shared_users"] = {"shared_users": shared_users}
-        filt["$or"] = [{"is_private": False}]
+        permissions_list = [{"is_private": False}]
         if get_user().is_authenticated:
             email = get_user().email
-            filt["$or"].append({"creator": email})
-            filt["$or"].append({"shared_users": email})
+            permissions_list.append({"creator": email})
+            permissions_list.append({"shared_users": email})
+
+        if not dataset_list:
```

Comment:
This special treatment of permissions when `dataset_list` is specified is a bit awkward, although I see why you had to do it this way. As an alternative suggestion. What about we do something like this at the beginning:
```
search_conditions = []
if XXX:
  search_conditions.append(XXX)
if YYY:
  search_conditions.append(YYY)
...
if dataset_list:
  ...
  search_conditions.append({"$or": dataset_dicts})
if get_user().is_authenticated:
  ...
  search_conditions.append({"$or": permissions_list})
filt = {"$and": search_conditions}
```